hkm/views/views.py

```diff
@@ -200,9 +202,48 @@ def get_collection_qs(self, request, *args, **kwargs):
         return Collection.objects.filter(owner=request.user)
 
 
+def ensure_collections_records_are_in_cache(request, collection_id):
+    """Fetch the given collection's Records from Finna and store them
+    in the cache, if they are not already found in the cache.
+
+    Fetching from Finna is done in multiple calls in case the Collection
+    contains a lot of Records. This is done to avoid a 414 error from Finna.
+    """
+    record_ids_in_collection = Collection.objects. \
+        user_can_view(request.user). \
+        get(id=collection_id). \
+        records.all(). \
+        values_list('record_id', flat=True)
+
+    record_ids_not_in_cache = []
+    for id in record_ids_in_collection:
+        cache_key = '%s-details' % id
+        if DEFAULT_CACHE.get(cache_key) is None:
+            record_ids_not_in_cache.append(id)
+
+    if record_ids_not_in_cache:
+        chunked_ids = list(chunks(record_ids_not_in_cache, 200))
+
+        i = 0
+        for chunk in chunked_ids:
+            records_from_finna = FINNA.get_record(chunk)
+
+            if records_from_finna and 'records' in records_from_finna:
+                for finna_record in records_from_finna['records']:
+                    cache_key = '%s-details' % finna_record['id']
+                    DEFAULT_CACHE.set(cache_key, finna_record, 60 * 15)
```

Comment:
Using a "constant" (like there is `RESULTS_PER_PAGE` for example) for the timeout value instead of these magic numbers in the middle of the code would be nicer IMO. (Yes I know rest of the codebase does not do that 🙂 )

Reply:
Oof, how did I not do it like that? Anyway, I added a constant now.